explainability/explainability-core/src/main/java/org/kie/kogito/explainability/utils/MatrixUtils.java

```diff
@@ -193,25 +197,38 @@ private static int findPivot(double[][] x, boolean[] pivotsUsed) {
             } catch (ArithmeticException e) {
                 // if the inversion is unsuccessful, we can try slightly jittering the matrix.
                 // this will reduce the accuracy of the inversion marginally, but ensures that we get results
-                MatrixUtils.jitterMatrix(x, 1e-8);
+                MatrixUtils.jitterMatrix(x, 1e-8, random);
             }
         }
 
         // if jittering didn't work, throw an error
         throw new ArithmeticException("Matrix is singular and could not be inverted via jittering");
     }
 
+    /**
+     * Jitter invert, but with a SecureRandom generator
+     *
+     * @param x a square double[][]; the matrix to be inverted
+     * @param numRetries the number of times to attempt jittering before giving up
+     * @return double[][], the inverted matrix
+     * @param zeroThreshold: the threshold to set such that x==0 if abs(x)<zeroThreshold. Use this avoid fp errors
+     *
+     */
+    public static double[][] jitterInvert(double[][] x, int numRetries, double zeroThreshold) {
+        return jitterInvert(x, numRetries, zeroThreshold, new SecureRandom());
+    }
+
```

Comment:
```suggestion
    /**
     * Jitter invert, but with a Random generator
     *
     * @param x a square double[][]; the matrix to be inverted
     * @param numRetries the number of times to attempt jittering before giving up
     * @return double[][], the inverted matrix
     * @param zeroThreshold: the threshold to set such that x==0 if abs(x)<zeroThreshold. Use this avoid fp errors
     *
     */
    public static double[][] jitterInvert(double[][] x, int numRetries, double zeroThreshold) {
        return jitterInvert(x, numRetries, zeroThreshold, new Random());
    }

```

nitpick.. afaik `SecureRandom` is ~10 times slower than `Random`. Since this is not security relevant, what about using `Random`?

Reply:
The original implementation used `Random`, but SonarCloud complains about a security hotspot if you initialize a non-secure random in a method, so I was asked by Daniele to make the default a SecureRandom. Since we allow the caller specify whichever random generator they want to actually use, whenever we use the method we'll pass a normal `Random`.